Dockerfile

```
@@ -1,22 +1,43 @@
 FROM golang:1.22-alpine3.19 AS builder
 
-RUN apk add --no-cache make
+# Create a non-root user
+RUN addgroup -g 1001 -S iamgroup && \
+	adduser -u 1001 -S iamuser -G iamgroup
 
-RUN mkdir -p /go/src/github.com/status-im/telemetry
 WORKDIR /go/src/github.com/status-im/telemetry
-ADD . .
-RUN make build
 
-# Copy the binary to the second image
+# Copy go mod and sum files for caching
+COPY go.mod go.sum ./
+
+# Using mounts to cache dependencies
+RUN --mount=type=cache,target=/go/pkg/mod/ \
+	--mount=type=cache,target=/root/.cache/go-build \
+	go mod download -x
```

Weird indent.
```suggestion
RUN --mount=type=cache,target=/go/pkg/mod/ \
    --mount=type=cache,target=/root/.cache/go-build \
    go mod download -x
```
Maybe you should use spaces.

---

This is weird, it was looking fine on my IDE, yea you are right, maybe I should change it to spaces.